src/nodes.coffee

```diff
@@ -1260,12 +1278,18 @@ exports.Call = class Call extends Base
     if @locationData and @needsUpdatedStartLocation
       @locationData.first_line = locationData.first_line
       @locationData.first_column = locationData.first_column
-      @locationData.range[0] = locationData.range[0]
+      @locationData.range = [
```

Comment:
What rule is that? This whole function is mutating `locationData`, just look at what it’s doing to `first_line` etc.

Reply:
So if we’re following the “immutable `locationData`“ rule, don’t we need a line `@locationData = {}`? Like

```coffee
if @locationData and @needsUpdatedStartLocation
  @locationData = {}
  # ... then same as above
```
So that we don’t mutate the old object originally referenced by `@locationData`, in case it’s shared by another node.